bool dio1
bool dio2
bool dio3
